app/main.py

```diff
@@ -1,2 +1,57 @@
+from fractions import Fraction
+from typing import Hashable, Any
+
+
+class Node:
```

Comment:
This class better be defined inside of a Dictionary since it is a part of its implementation and will not be used outside of it
Also you should include hash_value as per task requirements